### PR TITLE
[TT-9752] Update NOTES.txt of each chart

### PR DESCRIPTION
**Pull request by buraksekili**
## Description

This PR updates NOTES of each chart

## Related Issue

https://tyktech.atlassian.net/browse/TT-9752

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tyk-stack/templates/NOTES.txt [hunk 1]
```diff
@@ -1,7 +1,39 @@
...
+    curl localhost:{{ index .Values "global" "servicePorts" "gateway" }}/hello
+{{- end }}
+
+{{ if .Values.global.components.bootstrap -}}
+=== Tyk Dashboard Login Details ===
+{{- if not .Values.global.adminUser.useSecretName }}
+    export TYK_DASHBOARD_ADMINEMAIL={{ .Values.global.adminUser.email }}
+    export TYK_DASHBOARD_ADMINPASSWORD=$(kubectl get secret --namespace {{ .Release.Namespace }} tyk-dashboard-login-details -o jsonpath="{.data.adminUserPassword}" | base64 --decode)
+{{- else }}
+    export TYK_DASHBOARD_ADMINEMAIL=$(kubectl get secret --namespace {{ .Release.Namespace }} {{ .Values.global.adminUser.useSecretName }} -o jsonpath="{.data.adminUserFirstName}" | base64 --decode)
```
Comment (komalsukhani) [inline tyk-stack/templates/NOTES.txt:21]:
```suggestion
    export TYK_DASHBOARD_ADMINEMAIL=$(kubectl get secret --namespace {{ .Release.Namespace }} {{ .Values.global.adminUser.useSecretName }} -o jsonpath="{.data.adminUserEmail}" | base64 --decode)
```

File: tyk-stack/templates/NOTES.txt [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
+    export TYK_DASHBOARD_ADMINPASSWORD=$(kubectl get secret --namespace {{ .Release.Namespace }} {{ .Values.global.adminUser.useSecretName }} -o jsonpath="{.data.adminUserPassword}" | base64 --decode)
+{{- end }}
+{{- end }}
+
+{{- if index .Values "tyk-dashboard" "dashboard" "ingress" "enabled" }}
+=== Ingress ===
+{{- range $host := .Values.dashboard.ingress.hosts }}
+  {{- range .paths }}
+  http{{ if $.Values.ingress.tls }}s{{ end }}://{{ $host.host }}{{ .path }}
```
Comment (komalsukhani) [inline tyk-stack/templates/NOTES.txt:30]:
```suggestion
  http{{ if $.Values.dashboard.ingress.tls }}s{{ end }}://{{ $host.host }}{{ .path }}
```

File: tyk-oss/templates/NOTES.txt [hunk 3]
```diff
@@ -1,21 +1,27 @@
...
+    kubectl port-forward --namespace {{ .Release.Namespace }} service/{{ $gwSvcName }} {{ index .Values "global" "servicePorts" "gateway" }}:{{ index .Values "global" "servicePorts" "gateway" }}
+    curl localhost:{{ index .Values "global" "servicePorts" "gateway" }}/hello
+{{- end }}
 
-To quickly test everything is ok, you can port-forward gateway pod to {{ index .Values "global" "servicePorts" "gateway" }} on your machine :
-    kubectl port-forward service/gateway-svc-{{ template "tyk-gateway.fullname" (index .Subcharts "tyk-gateway")}} {{ index .Values "global" "servicePorts" "gateway" }}:{{ index .Values "global" "servicePorts" "gateway" }}
-and run `curl localhost:{{ index .Values "global" "servicePorts" "gateway" }}/hello` to see some data back that includes redis info.
+=== API Secret (TYK_GW_SECRET) ===
+{{- if not .Values.global.secrets.useSecretName }}
+    export TYK_GW_SECRET={{ .Values.global.secrets.APISecret }}
```
Comment (komalsukhani) [inline tyk-oss/templates/NOTES.txt:18]: We should not print the secret in the notes.
We should give command to get it from k8s secret.

**Review by komalsukhani — CHANGES_REQUESTED** [review]

File: components/tyk-gateway/templates/NOTES.txt [hunk 4]
```diff
@@ -1,7 +1,18 @@
...
 At this point, Tyk Gateway is fully installed and should be accessible.
 
-You might want to install Tyk Operator next to manage Ingress resources or manage your APIs.
+To quickly test everything is ok, you can port-forward Tyk Gateway service:
+{{ if not .Values.gateway.control.enabled }}
+    kubectl port-forward service/{{- $svcName}} --namespace {{ .Release.Namespace }} {{ .Values.global.servicePorts.gateway  }}:{{ .Values.global.servicePorts.gateway }}
+    curl http://localhost:{{ .Values.global.servicePorts.gateway }}/hello
+{{- else }}
+    kubectl port-forward service/gateway-control-svc-{{ include "tyk-gateway.fullname" . }} --namespace {{ .Release.Namespace }} {{ .Values.gateway.control.port }}:{{ .Values.gateway.control.port }}
+    curl http://localhost:{{ .Values.gateway.control.port }}
```
[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-gateway/templates/NOTES.txt:11]: yes, it is missing

[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-gateway/templates/NOTES.txt]: updated in 734f8a516fae1da9a8a4f834ce5aa7bfa7a3ceb3

File: components/tyk-dashboard/templates/NOTES.txt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+{{- if not .Values.global.adminUser.useSecretName }}
+    export TYK_DASHBOARD_ADMINEMAIL={{ .Values.global.adminUser.email }}
+    export TYK_DASHBOARD_ADMINPASSWORD=$(kubectl get secret --namespace {{ .Release.Namespace }} tyk-dashboard-login-details -o jsonpath="{.data.adminUserPassword}" | base64 --decode)
+{{- else }}
+    export TYK_DASHBOARD_ADMINEMAIL=$(kubectl get secret --namespace {{ .Release.Namespace }} {{ .Values.global.adminUser.useSecretName }} -o jsonpath="{.data.adminUserFirstName}" | base64 --decode)
+    export TYK_DASHBOARD_ADMINPASSWORD=$(kubectl get secret --namespace {{ .Release.Namespace }} {{ .Values.global.adminUser.useSecretName }} -o jsonpath="{.data.adminUserPassword}" | base64 --decode)
+{{- end }}
+
+{{- if .Values.dashboard.ingress.enabled }}
+=== Ingress ===
```
[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-dashboard/templates/NOTES.txt:22]: updated in 19b5b081584855b0978d3bb88c886eceb18da0e2

[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-dashboard/templates/NOTES.txt:22]: 19b5b081584855b0978d3bb88c886eceb18da0e2

File: components/tyk-bootstrap/templates/NOTES.txt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+Thank you for installing Tyk Bootstrap Chart.
+
+Tyk Bootstrap chart comes with Kubernetes Jobs to bootstrap
+- Tyk Dashboard
```
[the remark this answers is not in the document]
Comment (buraksekili) [reply components/tyk-bootstrap/templates/NOTES.txt]: addressed in https://github.com/TykTechnologies/tyk-charts/pull/186/commits/274429198828355c6ae21df7031b93e644570049

File: tyk-oss/templates/NOTES.txt (hunk 3 above)
Comment (buraksekili) [reply tyk-oss/templates/NOTES.txt]: updated accordingly, can you please check again?

File: tyk-oss/templates/NOTES.txt [hunk 7]
```diff
@@ -1,21 +1,28 @@
...
+    curl localhost:{{ index .Values "global" "servicePorts" "gateway" }}/hello
+{{- end }}
 
-To quickly test everything is ok, you can port-forward gateway pod to {{ index .Values "global" "servicePorts" "gateway" }} on your machine :
-    kubectl port-forward service/gateway-svc-{{ template "tyk-gateway.fullname" (index .Subcharts "tyk-gateway")}} {{ index .Values "global" "servicePorts" "gateway" }}:{{ index .Values "global" "servicePorts" "gateway" }}
-and run `curl localhost:{{ index .Values "global" "servicePorts" "gateway" }}/hello` to see some data back that includes redis info.
+=== API Secret (TYK_GW_SECRET) ===
+{{- if not .Values.global.secrets.useSecretName -}}
+{{ $gwFullName := (include "tyk-gateway.fullname" (index .Subcharts "tyk-gateway")) }}
+    export TYK_GW_SECRET=$(kubectl get secrets secrets-{{- $gwFullName }} -o jsonpath="{.data.APISecret}" | base64 --decode)
```
Comment (caroltyk) [inline tyk-oss/templates/NOTES.txt:19]: hi @burak, kubectl get secret needs to include --namespace too

File: tyk-stack/templates/NOTES.txt [hunk 8]
```diff
@@ -1,7 +1,49 @@
...
+    export TYK_DASHBOARD_ADMINPASSWORD=$(kubectl get secret --namespace {{ .Release.Namespace }} tyk-dashboard-login-details -o jsonpath="{.data.adminUserPassword}" | base64 --decode)
+{{- else }}
+    export TYK_DASHBOARD_ADMINEMAIL=$(kubectl get secret --namespace {{ .Release.Namespace }} {{ .Values.global.adminUser.useSecretName }} -o jsonpath="{.data.adminUserEmail}" | base64 --decode)
+    export TYK_DASHBOARD_ADMINPASSWORD=$(kubectl get secret --namespace {{ .Release.Namespace }} {{ .Values.global.adminUser.useSecretName }} -o jsonpath="{.data.adminUserPassword}" | base64 --decode)
+{{- end }}
+{{- end }}
+
+{{- if index .Values "tyk-dashboard" "dashboard" "ingress" "enabled" }}
+=== Ingress ===
+{{- range $host := .Values.dashboard.ingress.hosts }}
```
Comment (caroltyk) [inline tyk-stack/templates/NOTES.txt:38]: Error: template: tyk-stack/templates/NOTES.txt:38:26: executing "tyk-stack/templates/NOTES.txt" at <.Values.dashboard.ingress.hosts>: nil pointer evaluating interface {}.ingress

I got this error while installing

Comment (buraksekili) [reply tyk-stack/templates/NOTES.txt]: fixed in https://github.com/TykTechnologies/tyk-charts/pull/186/commits/5ac80e7d372523277935e01db24dfdd5333e86bc

File: tyk-oss/templates/NOTES.txt (hunk 7 above)
Comment (buraksekili) [reply tyk-oss/templates/NOTES.txt]: addressed in a40370348d63398f62303df6581184275590e6d6

**Pull request merged**